compaction/compaction_manager.cc

```diff
@@ -1878,6 +1878,11 @@ const std::unordered_set<sstables::shared_sstable>& compaction_manager::sstables
 }
 
 future<> compaction_manager::perform_cleanup(owned_ranges_ptr sorted_owned_ranges, table_state& t, std::optional<tasks::task_info> info) {
+    auto gh = start_compaction(t);
+    if (!gh) {
+        co_return;
```

Comment:
I'm not familiar with this code, but I looked at `start_compaction`
```cpp
std::optional<gate::holder> compaction_manager::start_compaction(table_state& t) {
    if (_state != state::enabled) {
        return std::nullopt;
    }

    auto it = _compaction_state.find(&t);
    if (it == _compaction_state.end() || it->second.gate.is_closed()) {
        return std::nullopt;
    }

    return it->second.gate.hold();
}
```

- `_state != state::enabled`: suppose that cleanup is requested from REST API, and instantly returns because we're in the middle of shutdown. The caller might interpret it as if cleanup was successful or not necessary. Shouldn't we instead return some kind of error / throw exception when someone/something tries to trigger cleanup in the middle of shutdown?
- `it == _compaction_state.end()` under what conditions could this happen?

Reply:
> Shouldn't we instead return some kind of error / throw exception when someone/something tries to trigger cleanup in the middle of shutdown?

This is consistent of all other usages of `start_compaction()`. Also, see below.

> it == _compaction_state.end() under what conditions could this happen?

The table-state (tablet or entire table) was removed. We don't want to raise an error in this case, just silently declare the task done.